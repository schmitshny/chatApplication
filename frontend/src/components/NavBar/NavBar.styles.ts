import styled from 'styled-components';

export const NavBarContainer = styled.nav`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  height: 100%;
  padding: 20px;
  padding-top: 100px;
  width: 60px;
`;

export const NavBarSection = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-items: center;
  gap: 40px;
`;

export const Logo = styled.div`
  font-size: 1.2em;
`;

export const UserContainer = styled.div`
  display: flex;
  align-items: center;
  align-content: center;
  cursor: pointer;
`;

export const UserName = styled.span`
  font-size: 1em;
  margin-left: 10px;
`;
