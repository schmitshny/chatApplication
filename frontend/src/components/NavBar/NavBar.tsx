import { useNavigate } from 'react-router-dom';
import { useAuthContext } from '../../features/auth/context/useAuthContext';
import { useNotifications } from '../../features/chat/useNotifications';
import { Icon } from '../Icon';
import { NavBarContainer, NavBarSection } from './NavBar.styles';

export const NavBar = () => {
  const { logout } = useAuthContext();
  const navigate = useNavigate();
  useNotifications();

  return (
    <NavBarContainer>
      <NavBarSection>
        <Icon name="chat" hoverText="Chats" onClick={() => navigate('/')} />
        <Icon name="media" hoverText="Stories" onClick={() => navigate('/stories')} />
        <Icon name="addContact" hoverText="Add Contact" onClick={() => navigate('/add-contact')} />
      </NavBarSection>
      <NavBarSection>
        <Icon name="notification" hoverText="Notifications" onClick={() => navigate('account/notifications')} />
        <Icon name="profile" hoverText="Account" onClick={() => navigate('/account')} />
        <Icon name="logout" hoverText="Log out" onClick={logout} customColor="#FF69B4" data-testid="logout-icon" />
      </NavBarSection>
    </NavBarContainer>
  );
};
