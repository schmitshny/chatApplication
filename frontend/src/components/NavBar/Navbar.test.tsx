import { describe, expect, test, vi } from 'vitest';
import { render } from '../../tests/test-utils';
import { NavBar } from './NavBar';

vi.mock('../../features/auth/context/useAuthContext', () => ({
  useAuthContext: vi.fn(() => ({
    logout: vi.fn(),
    user: null,
    isLoggedIn: false,
    setUserContext: vi.fn(),
  })),
}));

vi.mock('react-router-dom', () => ({
  useNavigate: () => vi.fn(),
}));

vi.mock('../../features/chat/useNotifications', () => ({
  useNotifications: vi.fn(),
}));

describe('NavBar Component', () => {
  test('renders NavBar correctly', () => {
    const { container } = render(<NavBar />);
    expect(container).toMatchSnapshot();
  });
});
