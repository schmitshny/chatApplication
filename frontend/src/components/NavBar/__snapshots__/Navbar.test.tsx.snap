// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar Component > renders NavBar correctly 1`] = `
<div>
  <nav
    class="sc-dkzCjb liKSgr"
  >
    <div
      class="sc-hKMtDE jevFpW"
    >
      <div
        class="sc-bczSft exNLyF"
      >
        <svg
          fill="none"
          height="48"
          viewBox="0 0 48 48"
          width="48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#1C1C1C"
            height="46"
            rx="13"
            stroke="url(#paint0_linear_21_125)"
            stroke-width="2"
            width="46"
            x="1"
            y="1"
          />
          <path
            d="M15.0632 12.486C17.533 12.1653 20.0511 12 22.6075 12C25.1636 12 27.6815 12.1653 30.151 12.4859C32.4394 12.783 34.0694 14.7016 34.2051 16.9225C33.8172 16.793 33.4062 16.7071 32.9762 16.6714C31.3232 16.5342 29.6516 16.4643 27.9642 16.4643C26.2769 16.4643 24.6054 16.5342 22.9524 16.6714C20.1449 16.9044 18.1429 19.2794 18.1429 21.9502V27.052C18.1429 29.0647 19.2794 30.9079 21.0391 31.795L17.8813 34.9527C17.626 35.2081 17.242 35.2845 16.9083 35.1463C16.5747 35.0081 16.3571 34.6825 16.3571 34.3214V29.5244C15.9244 29.4781 15.4931 29.427 15.0632 29.3712C12.6722 29.0607 11 26.9801 11 24.6341V17.2231C11 14.877 12.6722 12.7964 15.0632 12.486Z"
            fill="white"
          />
          <path
            d="M27.9642 18.25C26.3263 18.25 24.704 18.3179 23.1001 18.451C21.2676 18.6031 19.9286 20.1581 19.9286 21.9502V27.052C19.9286 28.8469 21.2714 30.4027 23.1062 30.5525C24.587 30.6734 26.0834 30.7395 27.5933 30.7488L30.9044 34.0599C31.1597 34.3152 31.5438 34.3916 31.8774 34.2534C32.211 34.1152 32.4286 33.7897 32.4286 33.4285V30.5834C32.5599 30.5736 32.6912 30.5633 32.8223 30.5526C34.6571 30.4029 36 28.847 36 27.0521V21.9503C36 20.1581 34.661 18.6031 32.8285 18.451C31.2245 18.3179 29.6022 18.25 27.9642 18.25Z"
            fill="white"
          />
          <defs>
            <lineargradient
              gradientUnits="userSpaceOnUse"
              id="paint0_linear_21_125"
              x1="24"
              x2="24"
              y1="0"
              y2="68"
            >
              <stop
                stop-color="#FFF500"
              />
              <stop
                offset="1"
                stop-color="#B42BA6"
              />
            </lineargradient>
          </defs>
        </svg>
        <span
          class="sc-gsnScm eXHkrn"
        >
          Chats
        </span>
      </div>
      <div
        class="sc-bczSft exNLyF"
      >
        <svg
          fill="none"
          height="27"
          viewBox="0 0 26 27"
          width="26"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M19.3411 0.633595C15.156 -0.211361 10.8442 -0.211361 6.65907 0.633595C5.35691 0.907776 4.15669 1.53878 3.19263 2.45605C2.22856 3.37331 1.53866 4.54068 1.20007 5.8276L1.13707 6.0536C0.284061 9.18691 0.284061 12.4913 1.13707 15.6246L1.19907 15.8506C1.5382 17.1366 2.22808 18.303 3.1917 19.2197C4.15532 20.1363 5.35474 20.7671 6.65607 21.0416C8.41249 21.3974 10.1958 21.6041 11.9871 21.6596V24.3876H9.63407C9.35772 24.3876 9.09268 24.4974 8.89727 24.6928C8.70186 24.8882 8.59207 25.1532 8.59207 25.4296C8.59207 25.706 8.70186 25.971 8.89727 26.1664C9.09268 26.3618 9.35772 26.4716 9.63407 26.4716H16.3661C16.6424 26.4716 16.9075 26.3618 17.1029 26.1664C17.2983 25.971 17.4081 25.706 17.4081 25.4296C17.4081 25.1532 17.2983 24.8882 17.1029 24.6928C16.9075 24.4974 16.6424 24.3876 16.3661 24.3876H14.0101V21.6616C15.8012 21.605 17.5845 21.3983 19.3411 21.0436C20.6431 20.7695 21.8432 20.1386 22.8073 19.2216C23.7714 18.3045 24.4613 17.1373 24.8001 15.8506L24.8631 15.6246C25.7161 12.4913 25.7161 9.18691 24.8631 6.0536L24.8001 5.8276C24.4615 4.54112 23.7718 3.37414 22.8081 2.45709C21.8445 1.54003 20.6447 0.909029 19.3431 0.634596L19.3411 0.633595ZM16.0101 12.5006L16.0441 12.4696L18.2531 10.4166C18.4122 10.2657 18.507 10.0594 18.518 9.84041C18.529 9.62143 18.4553 9.40663 18.3121 9.24059C18.2426 9.15961 18.1575 9.09354 18.0618 9.04636C17.9661 8.99917 17.8619 8.97186 17.7553 8.96607C17.6488 8.96027 17.5422 8.97612 17.4419 9.01264C17.3417 9.04917 17.2499 9.10562 17.1721 9.1786L14.9631 11.2306C14.8131 11.3733 14.6591 11.5116 14.5011 11.6456C14.3804 11.4756 14.2647 11.3019 14.1541 11.1246L13.1901 9.6306L13.1651 9.5926C13.0034 9.32824 12.8247 9.0747 12.6301 8.8336C12.4033 8.5436 12.0748 8.35058 11.7111 8.2936C11.3465 8.25576 10.9812 8.3588 10.6901 8.5816C10.4435 8.76565 10.2097 8.9661 9.99007 9.18159L9.95607 9.21259L7.74907 11.2636C7.58984 11.4143 7.49478 11.6206 7.4836 11.8396C7.47243 12.0586 7.54601 12.2734 7.68907 12.4396C7.75838 12.5208 7.84343 12.587 7.93909 12.6344C8.03475 12.6817 8.13902 12.7092 8.24559 12.7151C8.35216 12.721 8.45882 12.7052 8.55912 12.6687C8.65941 12.6321 8.75125 12.5756 8.82907 12.5026L11.0381 10.4516C11.2541 10.2516 11.3941 10.1226 11.5001 10.0366C11.5811 10.1456 11.6851 10.3056 11.8471 10.5556L12.8121 12.0556L12.8371 12.0936C12.9987 12.3573 13.1774 12.6102 13.3721 12.8506C13.5992 13.1404 13.9275 13.3337 14.2911 13.3916C14.6554 13.4281 15.0199 13.324 15.3101 13.1006C15.5559 12.9157 15.7897 12.7153 16.0101 12.5006Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="sc-gsnScm eXHkrn"
        >
          Stories
        </span>
      </div>
      <div
        class="sc-bczSft exNLyF"
      >
        <svg
          fill="none"
          height="41"
          viewBox="0 0 40 41"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 14.1279C15 11.8498 16.8468 10.0029 19.125 10.0029C21.4032 10.0029 23.25 11.8498 23.25 14.1279C23.25 16.4061 21.4032 18.2529 19.125 18.2529C16.8468 18.2529 15 16.4061 15 14.1279Z"
            fill="currentColor"
          />
          <path
            d="M12 26.8779C12 22.9429 15.19 19.7529 19.125 19.7529C23.06 19.7529 26.25 22.9429 26.25 26.8779V26.8805C26.25 26.9203 26.2496 26.9603 26.249 26.9999C26.2446 27.2584 26.1074 27.4964 25.8859 27.6298C23.9107 28.819 21.5966 29.5029 19.125 29.5029C16.6534 29.5029 14.3393 28.819 12.3641 27.6298C12.1426 27.4964 12.0054 27.2584 12.001 26.9999C12.0003 26.9593 12 26.9187 12 26.8779Z"
            fill="currentColor"
          />
          <path
            d="M28.5 15.2529C28.5 14.8387 28.1642 14.5029 27.75 14.5029C27.3358 14.5029 27 14.8387 27 15.2529V17.5029H24.75C24.3358 17.5029 24 17.8387 24 18.2529C24 18.6671 24.3358 19.0029 24.75 19.0029H27V21.2529C27 21.6671 27.3358 22.0029 27.75 22.0029C28.1642 22.0029 28.5 21.6671 28.5 21.2529V19.0029H30.75C31.1642 19.0029 31.5 18.6671 31.5 18.2529C31.5 17.8387 31.1642 17.5029 30.75 17.5029H28.5V15.2529Z"
            fill="currentColor"
          />
          <circle
            cx="20"
            cy="20.4716"
            r="19.5"
            stroke="currentColor"
            stroke-dasharray="5 5"
          />
        </svg>
        <span
          class="sc-gsnScm eXHkrn"
        >
          Add Contact
        </span>
      </div>
    </div>
    <div
      class="sc-hKMtDE jevFpW"
    >
      <div
        class="sc-bczSft exNLyF"
      >
        <svg
          fill="currentColor"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.85036 3.50019C6.1266 3.19155 6.10034 2.7174 5.79169 2.44116C5.48305 2.16491 5.0089 2.19118 4.73266 2.49982C3.5317 3.84163 2.69605 5.51948 2.3846 7.37592C2.31607 7.78442 2.59167 8.17114 3.00017 8.23967C3.40868 8.3082 3.79539 8.0326 3.86393 7.6241C4.12704 6.05581 4.83301 4.63686 5.85036 3.50019Z"
            fill="currentColor"
          />
          <path
            d="M19.267 2.49982C18.9908 2.19118 18.5166 2.16491 18.208 2.44116C17.8993 2.7174 17.873 3.19155 18.1493 3.50019C19.1666 4.63686 19.8726 6.05581 20.1357 7.6241C20.2043 8.0326 20.591 8.3082 20.9995 8.23967C21.408 8.17114 21.6836 7.78442 21.6151 7.37592C21.3036 5.51948 20.4679 3.84163 19.267 2.49982Z"
            fill="currentColor"
          />
          <path
            clip-rule="evenodd"
            d="M12 2.25C8.27215 2.25 5.25012 5.27197 5.25001 8.9998L5.24981 9.75C5.24981 11.8731 4.4488 13.8074 3.13126 15.2699C2.96476 15.4547 2.90073 15.71 2.96033 15.9516C3.01992 16.1931 3.19539 16.3893 3.42875 16.4755C4.97287 17.0455 6.58934 17.4659 8.2604 17.7192C8.25351 17.812 8.25001 17.9056 8.25001 18C8.25001 20.0711 9.92894 21.75 12 21.75C14.0711 21.75 15.75 20.0711 15.75 18C15.75 17.9056 15.7465 17.812 15.7396 17.7192C17.4105 17.4659 19.0269 17.0455 20.5709 16.4755C20.8042 16.3893 20.9797 16.1931 21.0393 15.9516C21.0989 15.71 21.0349 15.4547 20.8684 15.2699C19.5508 13.8074 18.7498 11.8731 18.7498 9.75V9.04919L18.75 9C18.75 5.27208 15.7279 2.25 12 2.25ZM9.75001 18C9.75001 17.9662 9.75075 17.9326 9.75221 17.8993C10.4927 17.966 11.2424 18 11.9998 18C12.7574 18 13.5072 17.9659 14.2478 17.8992C14.2493 17.9326 14.25 17.9662 14.25 18C14.25 19.2426 13.2427 20.25 12 20.25C10.7574 20.25 9.75001 19.2426 9.75001 18Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="sc-gsnScm eXHkrn"
        >
          Notifications
        </span>
      </div>
      <div
        class="sc-bczSft exNLyF"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 22 24"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M5.27194 5.14286C5.27194 2.30254 7.83648 0 11 0C14.1635 0 16.7281 2.30254 16.7281 5.14286C16.7281 7.98318 14.1635 10.2857 11 10.2857C7.83648 10.2857 5.27194 7.98318 5.27194 5.14286Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M0.500142 21.2632C0.598589 16.1321 5.26173 12 11 12C16.7384 12 21.4016 16.1323 21.4999 21.2636C21.5064 21.6039 21.288 21.9154 20.9435 22.0574C17.9153 23.3049 14.547 24 11.0004 24C7.45351 24 4.08487 23.3048 1.0565 22.057C0.711961 21.9151 0.493613 21.6035 0.500142 21.2632Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="sc-gsnScm eXHkrn"
        >
          Account
        </span>
      </div>
      <div
        class="sc-bczSft iewSQz"
        data-testid="logout-icon"
      >
        <svg
          fill="#FF69B4"
          height="27"
          id="Capa_1"
          version="1.1"
          viewBox="0 0 471.2 471.2"
          width="26"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="SVGRepo_bgCarrier"
            stroke-width="0"
          />
          <g
            id="SVGRepo_tracerCarrier"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
          <g
            id="SVGRepo_iconCarrier"
          >
             
            <g>
               
              <g>
                 
                <path
                  d="M227.619,444.2h-122.9c-33.4,0-60.5-27.2-60.5-60.5V87.5c0-33.4,27.2-60.5,60.5-60.5h124.9c7.5,0,13.5-6,13.5-13.5 s-6-13.5-13.5-13.5h-124.9c-48.3,0-87.5,39.3-87.5,87.5v296.2c0,48.3,39.3,87.5,87.5,87.5h122.9c7.5,0,13.5-6,13.5-13.5 S235.019,444.2,227.619,444.2z"
                />
                 
                <path
                  d="M450.019,226.1l-85.8-85.8c-5.3-5.3-13.8-5.3-19.1,0c-5.3,5.3-5.3,13.8,0,19.1l62.8,62.8h-273.9c-7.5,0-13.5,6-13.5,13.5 s6,13.5,13.5,13.5h273.9l-62.8,62.8c-5.3,5.3-5.3,13.8,0,19.1c2.6,2.6,6.1,4,9.5,4s6.9-1.3,9.5-4l85.8-85.8 C455.319,239.9,455.319,231.3,450.019,226.1z"
                />
                 
              </g>
               
            </g>
             
          </g>
        </svg>
        <span
          class="sc-gsnScm eXHkrn"
        >
          Log out
        </span>
      </div>
    </div>
  </nav>
</div>
`;
